components/core/tests/test-NetworkReader.cpp

```diff
@@ -193,29 +193,39 @@ TEST_CASE("network_reader_illegal_offset", "[NetworkReader]") {
 }
 
 TEST_CASE("network_reader_with_valid_http_header_kv_pairs", "[NetworkReader]") {
+    constexpr size_t cNumMaxTrials{10};
     std::unordered_map<std::string, std::string> valid_http_header_kv_pairs;
     // We use httpbin (https://httpbin.org/) to test the user-specified headers. On success, it is
     // supposed to respond all the user-specified headers as key-value pairs in JSON form.
-    constexpr int cNumHttpHeaderKeyValuePairs{10};
+    constexpr size_t cNumHttpHeaderKeyValuePairs{10};
     for (size_t i{0}; i < cNumHttpHeaderKeyValuePairs; ++i) {
         valid_http_header_kv_pairs.emplace(
                 fmt::format("Unit-Test-Key{}", i),
                 fmt::format("Unit-Test-Value{}", i)
         );
     }
-    clp::NetworkReader reader{
-            "https://httpbin.org/headers",
-            0,
-            false,
-            clp::CurlDownloadHandler::cDefaultOverallTimeout,
-            clp::CurlDownloadHandler::cDefaultConnectionTimeout,
-            clp::NetworkReader::cDefaultBufferPoolSize,
-            clp::NetworkReader::cDefaultBufferSize,
-            valid_http_header_kv_pairs
-    };
-    auto const content{get_content(reader)};
-    REQUIRE(assert_curl_error_code(CURLE_OK, reader));
-    auto const parsed_content = nlohmann::json::parse(content);
+    std::optional<std::vector<char>> content{std::nullopt};
+    // Retry the unit test a limited number of times to handle transient server-side HTTP errors.
+    // This ensures the test is not marked as failed due to temporary issues beyond our control.
+    for (size_t i{0}; i < cNumMaxTrials; ++i) {
+        clp::NetworkReader reader{
+                "https://httpbin.org/headers",
+                0,
+                false,
+                clp::CurlDownloadHandler::cDefaultOverallTimeout,
+                clp::CurlDownloadHandler::cDefaultConnectionTimeout,
+                clp::NetworkReader::cDefaultBufferPoolSize,
+                clp::NetworkReader::cDefaultBufferSize,
+                valid_http_header_kv_pairs
+        };
+        content.emplace(get_content(reader));
+        if (false == assert_curl_error_code(CURLE_OK, reader)) {
+            continue;
+        }
+        break;
```

Comment:
```suggestion
        if (assert_curl_error_code(CURLE_OK, reader)) {
            content.emplace(get_content(reader));
            break;
        }
```
- `content.emplace(get_content(reader))` should only execute if the return code is correct, otherwise the coming assertions on `content.has_value()` is meaningless.
- I think this if condition will make the code cleaner

Reply:
But I believe that it must call get_content first, otherwise the code in the reader won’t be set right?

Reply:
Right, but you can do this:
```suggestion
        auto content = get_content(reader);
        if (assert_curl_error_code(CURLE_OK, reader)) {
            optional_content.emplace(std::move(content));
            break;
        }
```